platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -209,240 +156,108 @@ private static void launchUnhandled(@NonNull final CommandLineArgs commandLineAr
-            @NonNull final MetricsProvider metricsProvider,
-            @NonNull final Configuration configuration,
-            @NonNull final InfoSwirld infoSwirld,
-            @NonNull final Crypto crypto)
-            throws IOException {
-
-        Objects.requireNonNull(nodeId);
-        Objects.requireNonNull(appDefinition);
-        Objects.requireNonNull(configAddressBook);
-        Objects.requireNonNull(appMain);
-        Objects.requireNonNull(metricsProvider);
-        Objects.requireNonNull(configuration);
-        Objects.requireNonNull(infoSwirld);
-        Objects.requireNonNull(crypto);
-
-        final Address address = configAddressBook.getAddress(nodeId);
-        final int instanceNumber = configAddressBook.getIndexOfNodeId(nodeId);
-
-        // this is a node to start locally.
-        final String platformName = address.getNickname()
-                + " - " + address.getSelfName()
-                + " - " + infoSwirld.getName()
-                + " - " + infoSwirld.getApp().getName();
-
-        final PlatformContext platformContext = new DefaultPlatformContext(nodeId, metricsProvider, configuration);
-
-        // name of the app's SwirldMain class
-        final String mainClassName = appDefinition.getMainClassName();
-        // the name of this swirld
-        final String swirldName = appDefinition.getSwirldName();
-        final SoftwareVersion appVersion = appMain.getSoftwareVersion();
-
-        final RecycleBinImpl recycleBin = new RecycleBinImpl(
-                configuration, platformContext.getMetrics(), getStaticThreadManager(), Time.getCurrent(), nodeId);
-        recycleBin.start();
-
-        // We can't send a "real" dispatch, since the dispatcher will not have been started by the
-        // time this class is used.
-        final BasicConfig basicConfig = configuration.getConfigData(BasicConfig.class);
-        final StateConfig stateConfig = configuration.getConfigData(StateConfig.class);
-        final EmergencyRecoveryManager emergencyRecoveryManager = new EmergencyRecoveryManager(
-                stateConfig, new Shutdown()::shutdown, basicConfig.getEmergencyRecoveryFileLoadDir());
-
-        final ReservedSignedState initialState;
-        try {
-            initialState = getInitialState(
-                    platformContext,
-                    recycleBin,
-                    appMain,
-                    mainClassName,
-                    swirldName,
-                    nodeId,
-                    configAddressBook,
-                    emergencyRecoveryManager);
-        } catch (final SignedStateLoadingException e) {
-            exitSystem(SystemExitCode.FATAL_ERROR, "unable to load initial state");
-            throw new IllegalStateException(); // unreachable
+    private static void startPlatforms(
+            @NonNull final List<SwirldsPlatform> platforms, @NonNull final Map<NodeId, SwirldMain> appMains) {
+
+        if (platforms.size() == 1) {
```

Comment:
Question: Why do we not start 1 platform in an external thread like we would do if there are 2? The current code makes a hard difference between having 1 or N platforms. Maybe not for this PR but from my point of view we should not have 2 different behaviors.

Reply:
I've switched this over to always using a thread. Mildly inefficient, but I'm ok making that sacrifice for easier to understand code.